clang/test/ClangScanDeps/gmodules.c

```diff
@@ -0,0 +1,228 @@
+// REQUIRES: ondisk_cas
+// REQUIRES: system-darwin
+
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: sed "s|DIR|%/t|g" %t/cdb.json.template > %t/cdb.json
+// RUN: sed "s|DIR|%/t|g" %t/cdb_pch.json.template > %t/cdb_pch.json
+// RUN: sed "s|DIR|%/t|g" %t/cas-config.template > %t/.cas-config
+
+/// Scan PCH
+// RUN: clang-scan-deps -compilation-database %t/cdb_pch.json \
+// RUN:   -cas-path %t/cas -module-files-dir %t/outputs \
+// RUN:   -format experimental-include-tree-full -mode preprocess-dependency-directives \
+// RUN:   > %t/deps_pch.json
+
+/// Build PCH
+// RUN: %deps-to-rsp %t/deps_pch.json --module-name Top > %t/Top.rsp
+// RUN: %deps-to-rsp %t/deps_pch.json --module-name Left > %t/Left.rsp
+// RUN: %deps-to-rsp %t/deps_pch.json --tu-index 0 > %t/pch.rsp
+// RUN: %clang @%t/Top.rsp
+// RUN: %clang @%t/Left.rsp
+// RUN: %clang @%t/pch.rsp
+
+/// Scan TU with PCH
+// RUN: clang-scan-deps -compilation-database %t/cdb.json \
+// RUN:   -cas-path %t/cas -module-files-dir %t/outputs \
+// RUN:   -format experimental-include-tree-full -mode preprocess-dependency-directives \
+// RUN:   > %t/deps.json
+
+// RUN: cat %t/deps.json | %PathSanitizingFileCheck --sanitize PREFIX=%/t %s
+
+/// Build TU
+// RUN: %deps-to-rsp %t/deps.json --module-name Right > %t/Right.rsp
+// RUN: %deps-to-rsp %t/deps.json --tu-index 0 > %t/tu.rsp
+// RUN: %clang @%t/Right.rsp
+// RUN: %clang @%t/tu.rsp
+
+/// Check debug info is correct.
+// RUN: %clang %t/tu.o -o %t/a.out
+// RUN: dsymutil -cas %t/cas %t/a.out -o %t/a.dSYM 2>&1 | FileCheck %s --check-prefix=WARN --allow-empty
+// RUN: dsymutil %t/a.out -o %t/a2.dSYM 2>&1 | FileCheck %s --check-prefix=WARN --allow-empty
```

Comment:
Does this being `dsymutil` and not `%dsymutil` mean we might pick up the system one? Since there is also a dsymutil aspect to this, could that be problematic in CI?

Reply:
Tools are always in the path.

NOTE: this is not the only test that calls dsymutil in clang tests and dsymutil wasn't even a build dependency for tests (added in this PR).